.github/actions/push-release-commit/action.yml

```diff
@@ -151,33 +140,25 @@ runs:
         
         gh api --method POST /repos/:owner/:repo/git/refs -f ref="refs/heads/$TEMPORARY_BRANCH" -f sha="$GITHUB_SHA"
 
-        # Parse modules and iterate over them, committing API files and collecting their details
-        # in a json object for later use
-        IFS=',' read -ra MOD_ARRAY <<< "$MODULES"
-
-        API_FILES_JSON="["
-
-        for mod in "${MOD_ARRAY[@]}"; do
-          MOD=$(echo "${mod}" | xargs)
-          API_FILE_PATH=$(git diff-tree --no-commit-id --name-only -r "$RELEASE_TAG" | grep "${MOD}/${API_FILE}")
-          API_FILE_INITIAL_SHA=$(git rev-parse "$GITHUB_REF":"$API_FILE_PATH")
-          API_FILE_RELEASE_SHA=$(git rev-parse "$RELEASE_TAG":"$API_FILE_PATH")
-          API_FILE_RELEASE_CONTENT=$(git cat-file blob "$RELEASE_TAG":"$API_FILE_PATH" | base64 -w0)
-          API_FILE_POST_RELEASE_CONTENT=$(git cat-file blob unsigned/"$GITHUB_REF_NAME":"$API_FILE_PATH" | base64 -w0)
-
-          # Commit the API file changes for module
-          gh api --method PUT /repos/:owner/:repo/contents/"$API_FILE_PATH" \
-            --field branch="$TEMPORARY_BRANCH" \
-            --field message="Update public API file for module $MOD for $RELEASE_TAG" \
-            --field sha="$API_FILE_INITIAL_SHA" \
-            --field content="$API_FILE_RELEASE_CONTENT" --jq '.commit.sha'
-          
-
-          API_FILES_JSON="${API_FILES_JSON}{\"module\":\"$MOD\",\"api_file_path\":\"$API_FILE_PATH\",\"api_file_release_sha\":\"$API_FILE_RELEASE_SHA\",\"api_file_post_release_content\":\"$API_FILE_POST_RELEASE_CONTENT\"},"
-        done
-
-        # Remove trailing comma and close JSON array
-        API_FILES_JSON="${API_FILES_JSON%,}]"
+        # Only commit API files for modules in API_UPDATED_MODULES
+        if [ -n "$API_UPDATED_MODULES" ]; then
```

Comment:
I'm not certain, but perhaps you don't need this `if [ -n "$API_UPDATED_MODULES" ]; then` conditional here - the [`-n`](https://www.bashsupport.com/bash/conditional/unary-operators/n/#:~:text=true%20if%20the%20string%20is,if%20a%20string%20is%20empty.) is just checking whether `API_UPDATED_MODULES` is a non-empty string - but then you're just iterating on each line of it with `IFS=',' read -ra UPDATED_MODS <<< "$API_UPDATED_MODULES"`, so if it has zero lines no work would get done anyway and there should be no crash, is that right?

Reply:
I wasn't certain about the behaviour of the next two lines:

```sh
          IFS=',' read -ra UPDATED_MODS <<< "$API_UPDATED_MODULES"
          for mod in "${UPDATED_MODS[@]}"; do
```

If I provide an empty string, my expectation was that the first line would make an array with a single entry of an empty string. And then the loop would run with `MOD` set to an empty string.

This was me just being cautious and bailing early if the incoming string was empty.